### ISSUE TITLE
"Pixel perfect tile object selection" should work on semi-transparent areas as well

### ISSUE THREAD

**Issue by user72356** [mapeditor/tiled#3404]
**Is your feature request related to a problem? Please describe.**
In https://github.com/mapeditor/tiled/issues/1477, you implemented an improved object selection mechanism that ignores transparent pixels so that one can select objects positioned below the transparent areas of other objects. I still have issues selecting objects below semi-transparent areas, for example shadows, fog, windows, etc. See attached screenshot for an example. 

**Describe the solution you'd like**
Simple modification to your existing code, change the alpha pixel check to consider anything less than 100% opaque as transparent for the purpose of selecting objects.

![image](https://user-images.githubusercontent.com/45011262/177202072-d1b3c1a3-6c77-4176-9672-5f75d0590257.png)

**Comment by eishiya** [comment]
Treating anything less than 100% opaque would be a problem for objects that are entirely or largely translucent, such as lamp glows, glass, gradients, etc. Some users may want to be able to click an object's shadow to select it, too. Different tilesets (and users!) have different cutoffs for what's appropriate to count as transparent.

Perhaps the pixel-perfect object selection checkbox should have a field next to it where users can specify the maximum alpha at which a pixel counts as transparent, which would default to 0 to match the current behaviour, but can be changed to a larger value if one wants.

**Comment by user72356** [comment]
Actually I have a better idea... We should be able to cycle between each object under the cursor on subsequent clicks on the same location. Like many software that deal with layers in fact.

**Comment by eishiya** [comment]
I haven't seen any software that do that, but it seems like a decent idea. If implemented, it should be optional to prevent accidental changes.

Also, in case you're unaware: if you alt+right click on a spot with multiple objects while using the Select Objects Tool, you'll get a menu with all the objects in that spot listed, which is very useful if the objects are named.

**Comment by bjorn** [comment]
I think you can cycle by holding Alt while left clicking, and indeed there is an Alt right click menu too.

**Comment by user72356** [comment]
Good point on the Alt+left click and Alt+right click. It's good that it's cycling! Unfortunately it doesn't let me grab and move the selected object, even if it appears to be the current selection. See the animated screenshot below. 

Would you consider making the cycling the default behavior rather than needing Alt? I think it qualifies as a non-disrupting enough... (and of course we should be able to move the selected object)

**Comment by eishiya** [comment]
Disagree on plain clicking cycling objects as a default, though it would be a fine option in Preferences. I am generally against features where repeating the same input does different things being default because it can be confusing and because it's very easy to accidentally cycle if you have jittery fingers/mouse.
The fact that the various mouse buttons and modifiers aren't customizable is one of Tiled's big usability gaps in my opinion, and this is a fine demonstration of that.

That said, I think that if you drag with some Objects selected, you pretty much always want to move those objects and not make a new selection, so I think some changes to what initiates a new selection might be good - though it would require some logic regarding what amount of movement still counts as a click rather than a drag. It would remove the need to hold Alt while moving lower objects, too, leaving Alt only required to select them.
